src/ucp/wireup/wireup.c

```diff
@@ -479,25 +487,8 @@ unsigned ucp_wireup_eps_progress(void *arg)
     /* Move wireup pending queue to temporary queue and remove references to
      * the wireup progress function
      */
-    ucs_queue_head_init(&tmp_pending_queue);
-    for (lane = 0; lane < ucp_ep_num_lanes(ucp_ep); ++lane) {
-        wireup_ep = ucp_wireup_ep(ucp_ep_get_lane(ucp_ep, lane));
-        if (wireup_ep == NULL) {
-            continue;
-        }
-
-        ucs_assert(wireup_ep->flags & UCP_WIREUP_EP_FLAG_READY);
-        ucs_assert(wireup_ep->super.uct_ep != NULL);
-
-        ucs_trace("ep %p: switching wireup_ep %p to ready state", ucp_ep,
-                  wireup_ep);
-        pending_count = ucp_wireup_ep_pending_extract(wireup_ep,
-                                                      &tmp_pending_queue);
-        ucp_worker_flush_ops_count_add(ucp_ep->worker, -pending_count);
-
-        /* Switch to real transport and destroy proxy endpoint (aux_ep as well) */
-        ucp_proxy_ep_replace(&wireup_ep->super);
-    }
+    ucp_wireup_eps_pending_extract(ucp_ep, ucp_wireup_proxy_ep_replace,
```

Comment:
why need to pass ucp_wireup_proxy_ep_replace as a callback, can't we go over the lanes after calling ucp_wireup_eps_pending_extract and call ucp_wireup_proxy_ep_replace for each lane?
in other places the callback seems to be empty function anyway

Reply:
We can do it. I thought to reduce iteration by using a CB.
In the rest of the calls it's indeed redundant at least for now.

Reply:
performance is not important here as much as making code simpler